lib/helpers/dart_syntax_highlighter.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart' show Color, TextSpan, TextStyle, Theme;
+import 'package:flutter_markdown/flutter_markdown.dart';
+import 'package:highlight/highlight.dart';
+
+class DartSyntaxHighlighter extends SyntaxHighlighter {
+  DartSyntaxHighlighter(this.context);
+
+  final String _rootKey = 'root';
+  final BuildContext context;
+
+  @override
+  TextSpan format(String source) {
+    return TextSpan(
+      style: _syntaxTheme[_rootKey],
+      children: _convert(
+        highlight.parse(source, language: 'dart').nodes!,
+      ),
+    );
+  }
+
+  List<TextSpan> _convert(List<Node> nodes) {
+    List<TextSpan> currentSpans = [];
+    final List<List<TextSpan>> stack = [];
+
+    void traverse(Node node) {
+      if (node.value != null) {
+        currentSpans.add(
+          node.className == null
+              ? TextSpan(text: node.value)
+              : TextSpan(text: node.value, style: _syntaxTheme[node.className]),
+        );
```

Comment:
```suggestion
        currentSpans.add(
          TextSpan(
            text: node.value,
            style: node.className != null ? _syntaxTheme[node.className] : null
          ),
        );
```

Reply:
Still, not really resolved since only the style property is different, I would put the conditional logic there. Anyway, not a biggie.

Reply:
oh, my bad i didn't notice the `null` check you added. I thought it was only line formatting